opm/output/eclipse/EclFilesComparator.cpp

```diff
@@ -0,0 +1,307 @@
+        OPM_THROW(std::runtime_error, "Keyword does not exist in second file.");
+    }
+    ecl_kw_type* ecl_kw = ecl_file_iget_named_kw(ecl_file1, keyword.c_str(), 0);
+    ecl_type_enum ecl_kw_type = ecl_kw_get_type(ecl_kw);
+    switch (ecl_kw_type) {
+        case ECL_DOUBLE_TYPE:
+            return true;
+        case ECL_FLOAT_TYPE:
+            return true;
+        case ECL_CHAR_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_CHAR_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        case ECL_INT_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_INT_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        case ECL_BOOL_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_BOOL_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        case ECL_MESS_TYPE:
+            std::cout << "\nKeyword " << keyword << " is of type ECL_MESS_TYPE, "
+                << "which is not supported for comparison." << std::endl;
+            return false;
+        default:
+            std::cout << "Keyword " << keyword << "has undefined type." << std::endl;
+            return false;
+    }
+}
+
+
+
+void ECLFilesComparator::printResultsForKeyword(const std::string keyword) const {
+    const double absDeviationAverage = average(absDeviation);
+    const double relDeviationAverage = average(relDeviation);
+    std::cout << "\nKeyword "                    << keyword              << ":\n\n";
+    std::cout << "Average absolute deviation = " << absDeviationAverage  << std::endl;
+    std::cout << "Median absolute deviation  = " << median(absDeviation) << std::endl;
+    std::cout << "Average relative deviation = " << relDeviationAverage  << std::endl;
+    std::cout << "Median relative deviation  = " << median(relDeviation) << "\n\n";
+}
+
+
+
+// This function finds grid data for a given report step, and passes them on to deviationsForCell
+void ECLFilesComparator::deviationsForReportStep(const std::string& keyword, int reportStep) {
+    ecl_kw_type* ecl_kw1               = ecl_file_iget_named_kw(ecl_file1, keyword.c_str(), reportStep);
+    ecl_kw_type* ecl_kw2               = ecl_file_iget_named_kw(ecl_file2, keyword.c_str(), reportStep);
+    const unsigned int numActiveCells1 = ecl_kw_get_size(ecl_kw1);
+    const unsigned int numActiveCells2 = ecl_kw_get_size(ecl_kw2);
+    if (numActiveCells1 != numActiveCells2) {
+        std::cout << "For keyword " << keyword << " and report step " << reportStep << ":\n";
+        OPM_THROW(std::runtime_error, "Number of active cells are different.");
+    }
+    std::vector<double> values1, values2; // Elements in the vector corresponds to active cells
+    values1.resize(numActiveCells1);
+    values2.resize(numActiveCells2);
```

Comment:
use the constructor, std::vector<double> values1(numActiveCells1), values2(numActiveCells2);